201-timeseriesinsights-environment-with-iothub/azuredeploy.json

```diff
@@ -0,0 +1,252 @@
+      "metadata": {
+        "description": "Name of the event source child resource. The name cannot include:   '<', '>', '%', '&', ':', '\\', '?', '/' and any control characters. All other characters are allowed."
+      }
+    },
+    "eventSourceDisplayName": {
+      "type": "string",
+      "defaultValue": "",
+      "maxLength": 90,
+      "metadata": {
+        "description": "An optional friendly name to show in tooling or user interfaces instead of the event source name."
+      }
+    },
+    "eventSourceTimestampPropertyName": {
+      "type": "string",
+      "defaultValue": "",
+      "maxLength": 90,
+      "metadata": {
+        "description": "The event property that will be used as the event source's timestamp. If a value isn't specified for timestampPropertyName, or if null or empty-string is specified, the event creation time will be used."
+      }
+    },
+    "eventSourceKeyName": {
+      "type": "string",
+      "defaultValue": "service",
+      "metadata": {
+        "description": "The name of the shared access key that the Time Series Insights service will use to connect to the event hub."
+      }
+    },
+    "accessPolicyReaderObjectIds": {
+      "type": "array",
+      "defaultValue": [],
+      "metadata": {
+        "description": "A list of object ids of the users or applications in AAD that should have Reader access to the environment. The service principal objectId can be obtained by calling the Get-AzureRMADUser or the Get-AzureRMADServicePrincipal cmdlets. Creating an access policy for AAD groups is not yet supported."
+      }
+    },
+    "accessPolicyContributorObjectIds": {
+      "type": "array",
+      "defaultValue": [],
+      "metadata": {
+        "description": "A list of object ids of the users or applications in AAD that should have Contributor access to the environment. The service principal objectId can be obtained by calling the Get-AzureRMADUser or the Get-AzureRMADServicePrincipal cmdlets. Creating an access policy for AAD groups is not yet supported."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    }
+  },
+  "variables": {
+    "environmentTagsOptions": [
+      null,
+      {
+        "displayName": "[parameters('environmentDisplayName')]"
+      }
+    ],
+    "environmentTagsValue": "[variables('environmentTagsOptions')[length(take(parameters('environmentDisplayName'), 1))]]",
+    "eventSourceTagsOptions": [
+      null,
+      {
+        "displayName": "[parameters('eventSourceDisplayName')]"
+      }
+    ],
+    "eventSourceTagsValue": "[variables('eventSourceTagsOptions')[length(take(parameters('eventSourceDisplayName'), 1))]]",
+    "eventSourceResourceId": "[if(empty(parameters('existingIoTHubResourceId')), resourceId('Microsoft.Devices/IotHubs', parameters('iotHubName')), parameters('existingIoTHubResourceId'))]"  },
```

Comment:
when you add the new params and the defaultValues, this can be simplified to:

resourceId(parameters('iotHubResourceGroup'), 'Microsoft.Devices/IotHubs', parameters('iotHubName')))]